File: src/r/pe_0009.r
# FUNCTION


is_triplet <- function(a, b, c) {
    (a < b) && (b < c) && (a ** 2 + b ** 2 == c ** 2)
}


# SCRIPT


for (a in 1:1000) {
    for (b in (a + 1):(1000 - a)) {
        c = 1000 - (a + b)
        if (is_triplet(a, b, c)) {
            message(a, b, c)
        }
    }
}

File: src/java/pe_0009.java
public class pe_0009 {

    // SCRIPT


    public static void main(String[] args) {
        int a, b, c;
    
        for (a = 1; a < 999; a++) {
            for (b = a + 1; b < (1000 - a); b++) {
                c = 1000 - (a + b);
                if (a * a + b * b == c * c) {
                    System.out.printf("%d %d %d\n", a, b, c);
                    System.exit(0);
                }
            }
        }
    }

}
